### PR TITLE
Enable dd-solvated TDSCF

### PR DESCRIPTION
**Pull request by mfherbst**
## Description

Extends DDX interface to support SCF response and TDSCF.

## Checklist
- [X] Tests added for any new features
- [X] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [X] Ready for review
- [X] Ready for merge

### CODE REVIEW

**Review by maxscheurer — APPROVED** [review]
Neat! Glad the `cpscf` interface seems to work with more and more models 😄

**Comment by mfherbst** [comment]
I don't really see if the CI failures are related to my changes.

@loriab Any ideas?

**Comment by JonathonMisiewicz** [comment]
The problems are unrelated to this PR. Rebase and push, and the problems will be fixed.

I'll review this PR later today, since I know the TDSCF code the best, nowadays.

File: doc/sphinxman/source/ddx.rst [hunk 1]
```diff
@@ -111,10 +111,11 @@ and further parameters allow to influence details of discretisation,
 numerical integration and iterative solvers,
 see the next sections for details.
 
-.. note:: At present dd-based solvation models
-          can only be used for energy calculations with SCF
-          wavefunctions. All ERI algorithms (``PK``, ``OUT_OF_CORE``, ``DIRECT``, ``DF``,
-          ``CD``) are supported.
+.. note:: At present PCM can only be used for energy calculations with SCF
```
Comment (JonathonMisiewicz) [inline doc/sphinxman/source/ddx.rst:114]:
```suggestion
.. note:: At present, PCM can only be used for energy calculations with SCF
```

**Review by JonathonMisiewicz — APPROVED** [review]

**Comment by loriab** [comment]
The Win failures are the pydantic 1.10.7 + typing extensions 4.6 we were seeing yesterday. It should have cleared up, some I'm retriggering one to check.

**Comment by mfherbst** [comment]
Thanks @JonathonMisiewicz  and @loriab !

**Pull request merged**